hearth/lib/hearth/http/fields.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Hearth
+  module HTTP
+    # Provides Hash like access for Headers and Trailers with key normalization
+    class Fields
+      include Enumerable
+
+      # @param [Hash<String,Field>] fields
+      def initialize(fields = {}, encoding: 'utf-8')
+        @entries = {}
+        fields.each_pair do |key, value|
+          self[key] = value
+        end
+        @encoding = encoding
```

Comment:
How do we actually use the encoding?  Do we need to apply the encoding at some point - either when taking a value in the  `[]=` method, or maybe more likely, in the `value` method of `Field`, eg:
```
def value
@values.compact.map { |v| escape_value(v) }.join(',').encode(@encoding)
end
```